db/postgresql-instance.yml

```diff
@@ -0,0 +1,219 @@
+    ConstraintDescription: >-
+      must begin with a letter and contain only alphanumeric characters, dash,
+      or underscore.
+  DBUser:
+    NoEcho: true
+    Description: The database admin account username
+    Type: String
+    MinLength: 1
+    MaxLength: 16
+    AllowedPattern: ^[a-zA-Z][a-zA-Z0-9_]*$
+    ConstraintDescription: must begin with a letter and contain only alphanumeric characters or underscore.
+  DBPassword:
+    NoEcho: true
+    Description: The database admin account password
+    Type: String
+    MinLength: 1
+    MaxLength: 41
+    AllowedPattern: ^[a-zA-Z0-9]+$
+    ConstraintDescription: must contain only alphanumeric characters.
+  DBAllocatedStorage:
+    Default: 20
+    Description: The size of the database (Gb)
+    Type: Number
+    MinValue: 20
+    MaxValue: 16384
+    ConstraintDescription: must be between 5 and 16384Gb.
+  DBInstanceClass:
+    Description: The database instance type
+    Type: String
+    Default: db.t2.small
+    ConstraintDescription: must select a valid database instance type.
+  DBSnapshot:
+    Default: ""
+    Description: >-
+      optional, the DB snapshot that is used to restore the DB instance,
+      not used in conjunction with CreateDBName
+    Type: String
+    AllowedPattern: ^[a-zA-Z]*[a-zA-Z0-9\-:]*$
+    ConstraintDescription: >-
+      must begin with a letter and contain only alphanumeric characters, dash,
+      or colon.
+  EnvironmentType:
+    Type: String
+    AllowedValues:
+      - Staging
+      - Production
+  VPCStack:
+    Description: VPC stack to add this cache into
+    Type: String
+    Default: DataVPCStack
+  CloudWatchWarnDiskSpace:
+    Default: 0
+    Description: >-
+      amount of free disk space in bytes that will trigger warning, 0 means no alarm
+    Type: Number
+  CloudWatchWarnCPU:
+    Default: 0
+    Description: percent of CPU that will trigger warning, 0 means no alarm
+    Type: Number
+    MinValue: 0
+    MaxValue: 100
+    ConstraintDescription: must be between 0 and 100
+  CloudWatchWarnConnections:
+    Default: 0
+    Description: amount of connections that will trigger warning, 0 means no alarm
+    Type: Number
+  OpsWarnSnsTopic:
+    Default: ""
+    Description: ops warn SNS Topic arn
+    Type: String
+  EngineVersion:
+    Description: PostgreSQL Engine Version 
+    Type: String
+Conditions:
+  UseSnapshot: !Not [!Equals [!Ref DBSnapshot, ""]]
+  CreateDB: !Not [!Equals [!Ref CreateDBName, ""]]
+  NamedInstance: !Not [!Equals [!Ref DBInstance, ""]]
+  CreateCloudWatchWarnDiskSpace: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnDiskSpace, 0]]
+  CreateCloudWatchWarnCPU: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnCPU, 0]]
+  CreateCloudWatchWarnConnections: !And
+    - !Not [!Equals [!Ref OpsWarnSnsTopic, ""]]
+    - !Not [!Equals [!Ref CloudWatchWarnConnections, 0]]
+Resources:
+  RDSSubnetGroup:
```

Comment:
This resource type [supports tags](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbsubnet-group.html#cfn-rds-dbsubnetgroup-tags); good to always include them when available